app/controllers/ManageGroupController.scala

```diff
@@ -46,25 +50,96 @@ class ManageGroupController @Inject()
 
   def showManageGroups: Action[AnyContent] = Action.async { implicit request =>
     isAuthorisedAgent { arn =>
-      val eventuallySummaries = for {
-        response <- agentPermissionsConnector.groupsSummaries(arn)
-      } yield response
+      isOptedIn(arn) { _ =>
+        val eventuallySummaries = for {
+          response <- agentPermissionsConnector.groupsSummaries(arn)
+        } yield response
 
-      eventuallySummaries.map { summaries: Option[(Seq[GroupSummary], Seq[DisplayClient])] =>
-        Ok(dashboard(summaries.getOrElse((Seq.empty[GroupSummary], Seq.empty[DisplayClient]))))
+        eventuallySummaries.map { summaries: Option[(Seq[GroupSummary], Seq[DisplayClient])] =>
+          Ok(dashboard(summaries.getOrElse((Seq.empty[GroupSummary], Seq.empty[DisplayClient]))))
+        }
       }
     }
   }
 
   def showManageGroupClients(groupId: String): Action[AnyContent] = Action.async { implicit request =>
     isAuthorisedAgent { arn =>
-      Ok(s"showManageGroupClients not yet implemented ${groupId}").toFuture
+      isOptedIn(arn) { _ =>
+        Ok(s"showManageGroupClients not yet implemented ${groupId}").toFuture
+      }
     }
   }
 
   def showManageGroupTeamMembers(groupId: String): Action[AnyContent] = Action.async { implicit request =>
     isAuthorisedAgent { arn =>
-      Ok(s"showManageGroupTeamMembers not yet implemented ${groupId}").toFuture
+      isOptedIn(arn) { _ =>
+        Ok(s"showManageGroupTeamMembers not yet implemented ${groupId}").toFuture
+      }
+    }
+  }
+
+  def showRenameGroup(groupId: String): Action[AnyContent] = Action.async { implicit request =>
+    isAuthorisedAgent { arn =>
+      isOptedIn(arn) { _ =>
+        val eventuallyGroup = for {
```

Comment:
you don't need this val because you can do all that follows in the Yield

Reply:
removed that for comprehension 👍